system_tests/storage_trie_test.go

```diff
@@ -0,0 +1,85 @@
+// Copyright 2025, Offchain Labs, Inc.
+// For license information, see:
+// https://github.com/OffchainLabs/nitro/blob/master/LICENSE.md
+
+package arbtest
+
+import (
+	"context"
+	"math/big"
+	"testing"
+	"time"
+
+	"github.com/ethereum/go-ethereum/core/rawdb"
+
+	"github.com/offchainlabs/nitro/util/arbmath"
+	"github.com/offchainlabs/nitro/validator/valnode"
+)
+
+var withL1 = true
+
+func TestStorageTrie(t *testing.T) {
+	t.Parallel()
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	builder := NewNodeBuilder(ctx).DefaultConfig(t, withL1)
+
+	// This test tests validates blocks at the end.
+	// For now, validation only works with HashScheme set.
+	builder.execConfig.Caching.StateScheme = rawdb.HashScheme
+	builder.nodeConfig.BlockValidator.Enable = false
+	builder.nodeConfig.Staker.Enable = true
+	builder.nodeConfig.BatchPoster.Enable = true
+	builder.nodeConfig.ParentChainReader.Enable = true
+	builder.nodeConfig.ParentChainReader.OldHeaderTimeout = 10 * time.Minute
+
+	valConf := valnode.TestValidationConfig
+	valConf.UseJit = true
+	_, valStack := createTestValidationNode(t, ctx, &valConf)
+	configByValidationNode(builder.nodeConfig, valStack)
+
+	cleanup := builder.Build(t)
+	defer cleanup()
+
+	ownerTxOpts := builder.L2Info.GetDefaultTransactOpts("Owner", ctx)
+	_, bigMap := builder.L2.DeployBigMap(t, ownerTxOpts)
+
+	// Store enough values to use just over 32M gas
+	toAdd := big.NewInt(1420)
+	// In the first transaction, don't clear any values.
+	toClear := big.NewInt(0)
+
+	userTxOpts := builder.L2Info.GetDefaultTransactOpts("Faucet", ctx)
+	tx, err := bigMap.ClearAndAddValues(&userTxOpts, toClear, toAdd)
+	Require(t, err)
+
+	receipt, err := builder.L2.EnsureTxSucceeded(tx)
+	Require(t, err)
+	tx1BlockNum := receipt.BlockNumber.Uint64()
+
+	want := uint64(32_015_781)
+	if receipt.GasUsed != want {
+		t.Errorf("Want GasUsed: %d: got: %d", want, receipt.GasUsed)
```

Comment:
The exact gas used number seems finicky, so I'd just check that `receipt.GasUsed - receipt.GasUsedForL1 > 20_000_000` or something like that.

Reply:
Done.